tap_google_ads/streams.py

```diff
@@ -743,31 +742,34 @@ def sync(self, sdk_client, customer, stream, config, state, query_limit):
         if selected_fields == {'segments.date'}:
             raise Exception(f"Selected fields is currently limited to {', '.join(selected_fields)}. Please select at least one attribute and metric in order to replicate {stream_name}.")
 
-        query = create_report_query(resource_name, selected_fields, query_date, end_date)
-        LOGGER.info(f"Requesting {stream_name} data for {utils.strftime(query_date, '%Y-%m-%d')}.")
+        while query_date <= end_date:
+            query = create_report_query(resource_name, selected_fields, query_date)
+            LOGGER.info(f"Requesting {stream_name} data for {utils.strftime(query_date, '%Y-%m-%d')}.")
 
-        try:
-            response = make_request(gas, query, customer["customerId"], config)
-        except GoogleAdsException as err:
-            LOGGER.warning("Failed query: %s", query)
-            LOGGER.critical(str(err.failure.errors[0].message))
-            raise RuntimeError from None
+            try:
```

Comment:
Why do we need a loop here? Old code didn't have it.

Reply:
Initially, report_query was generated using `start_date` and `end_date` and whole data was fetched within range. But this query generation did not work for stream _click_performance_report_ which only provided day-wise data within 90 days interval.
Hence, Loop was introduced in the recent version where report data is fetched in day-wise manner for each stream considering the `start_date` provided by user till current data which becomes `end_date` and iteration is done using `query_date` variable.
Bookmark is saved after each sync and on next run sync starts in incremental way using saved state.